src/blog/system-analyzer.md

```diff
@@ -0,0 +1,204 @@
+---
+title: 'Indicium: V8 runtime tracer tool'
+author: 'Zeynep Cankara ([@zcankara](https://twitter.com/zeynepcankara))'
+avatars:
+  - 'zeynep-cankara'
+date: 2020-10-01 11:56:00
+tags:
+  - tools
+  - system-analyzer
+description: 'Indicium: V8 system analyzer tool to analyze Map/IC events.'
+---
+# Indicium: V8 system analyzer
+
+The past three  months have been an awesome learning experience for me as I've joined the V8 team (Google London) as an intern, and have been  working on a new tool called [*Indicium*](https://v8.dev/tools/system-analyzer).
+
+This system analyzer is a unified web interface to trace, debug and analyse patterns of how Inline Caches (ICs) and Maps are created and modified in real-world applications.
+
+V8 already has a tracing infrastructure for [ICs](https://mathiasbynens.be/notes/shapes-ics) and [Maps](https://v8.dev/blog/fast-properties) which can process and analyse IC events using the [IC Explorer](https://v8.dev/tools//ic-explorer.html) and Map events using [Map Processor](https://v8.dev/tools/map-processor.html). However, previous tools didn't allow us to analyze maps and ICs holistically and this is now possible with system analyzer .
+
+![Indicium](/_img/system-analyzer/indicium-logo.png)
+
+## Case Study
+
+Let’s go through an example to demonstrate how we can use the Indicium to analyse Map and IC log events in V8.
+
+```javascript
+class Point {
+ constructor(x, y) {
+   if (x < 0 || y < 0) {
+     this.isNegative = true;
+   }
+   this.x = x;
+   this.y = y;
+ }
+
+ dotProduct(other) {
+   return this.x * other.x + this.y * other.y;
+ }
+}
+
+let a = new Point(1, 1);
+let b = new Point(2, 2);
+let dotProduct;
+
+// warmup
+for (let i = 0; i < 10e5; i++) {
+  dotProduct = a.dotProduct(b);
+}
+
+start = performance.now();
```

Comment:
Btw, any reason you're not using `console.time` + `console.timeEnd` in those places? Looks like you're essentially trying to simulate them manually.